### PR TITLE
feat(activity): add activity 2

### PR DESCRIPTION
**Pull request by lwasser**
This is the second activity in the workshop. It's complete short of the answers to the OYO sections. i have an answer key script as well that i can push separately.

### CODE REVIEW

**Comment by lwasser** [comment]
NOTE: the CI issues will be fixed when some of the other pr's are merged. I'm worrying less about that now and more about content!!

File: clean-modular-code/activity-2/clean-code-activity-2.md
Comment (ucodery) [inline clean-modular-code/activity-2/clean-code-activity-2.md:31]:
```suggestion
* Document what the code does by adding a docstring at the top of the script to help your future self, a future lab mate, or another user understand the tasks that your code performs.
```

File: clean-modular-code/activity-2/clean-code-activity-2.md
Comment (ucodery) [inline clean-modular-code/activity-2/clean-code-activity-2.md:54]: I definitely thought you were leading up to Window vs POSIX path slashes. I think that is a more basic, and easier to automate, difference between environments. If the directory structure actually changes in new environments, I think there is only so much you can do without needing user input.

File: clean-modular-code/activity-2/clean-code-activity-2.md
Comment (ucodery) [inline clean-modular-code/activity-2/clean-code-activity-2.md:202]:
```suggestion
1. Create a workflow that calculates how many packages in that list are pyOpenSci packages. To calculate how many are pyOpenSci packages, you will need the `pyos-joss-accepted-packages.csv` file in the `data/` directory with the activity 2 JSON data. This .csv file lists all pyOpenSci packages that have also gone through JOSS review.
```

File: clean-modular-code/activity-2/clean-code-activity-2.md
Comment (lwasser) [reply clean-modular-code/activity-2/clean-code-activity-2.md:54]: Oh - @ucodery i didn't understand this comment at first but now i think i do. 

this section IS about windows vs POSIC path slashes. I think my language that says "directory structure on another machine differs" is confusing / incorrect!!

that is a much harder problem that we can't solve with code as you point out. How about i edit this paragraph to make it more clear. it's fully about path creation which we can handle with code.

**Comment by lwasser** [comment]
Cool - CI is failing with things are are fixed in #18 #13  i am going to merge this soon after one last round of revisions and adding answers to the OYOs

**Review by ucodery — APPROVED** [review]
I just read this one section, I couldn't easily tell if anything else has changed

File: clean-modular-code/activity-2/clean-code-activity-2.md
Comment (lwasser) [reply clean-modular-code/activity-2/clean-code-activity-2.md:54]: oh you're right! that's an important nit!!

**Comment by lwasser** [comment]
thank you, @ucodery. If you want to have a look at a preview of the published (ci) page - it's here - https://output.circle-artifacts.com/output/job/b18a4bf1-8f83-4764-a52b-e0e61a07f909/artifacts/0/html/clean-modular-code/activity-2/clean-code-activity-2.html I've made small edits and clarified things which is why it may be hard to identify what has changed.

The big change here is that I rewrote the POSIX path section and added examples. I'm hopeful that makes it more clear!  Thank you for the review comments - things were not worded well before.

**Comment by lwasser** [comment]
I am going to merge this PR, rebase and focus more time on activity 3 next week. Thank you again @ucodery for all of the input!! the CI issues are related to other lessons not in this PR. They will all get fixed once things are merged! Lesson dev is iterative as we focus on the content and move things around accordingly.

**Pull request merged**
